src/EA.Weee.RequestHandlers/AatfReturn/CopyReturnHandler.cs

```diff
@@ -62,9 +68,43 @@ public async Task<Guid> HandleAsync(CopyReturn message)
 
             await context.SaveChangesAsync();
 
+            await RemoveAatfsWithInvalidApprovalDate(returnCopy);
```

Comment:
Had issues copying the return as the ID's are not recreated until the context is saved so couldn't remove... but as you say there will be issues if errors occur after the first save. To resolve this will run in transaction.

Reply:
Updated to use transaction.